### PR TITLE
added parse-pe-export-data rule

### PR DESCRIPTION
**Pull request by sara-rn**
it detects manual parsing of the PE Export Data when resolving function RVAs dynamically.
lint.py fails to execute.

### CODE REVIEW

File: nursery/parse-pe-export-data.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        - 3 or more:
+          - offset: 0x14 = IMAGE_EXPORT_DIRECTORY.NumberOfFunctions
+          - offset: 0x24 = IMAGE_EXPORT_DIRECTORY.AddressOfNameOrdinals
+          - offset: 0x20 = IMAGE_EXPORT_DIRECTORY.AddressOfNames
+          - offset: 0x18 = IMAGE_EXPORT_DIRECTORY.NumberOfNames
+          - offset: 0x1C = IMAGE_EXPORT_DIRECTORY.AddressOfFunctions
+      - optional:
+        - or:
+          - api: LoadLibrary
+          - api: strcmp
```
Comment (mr-tz) [inline nursery/parse-pe-export-data.yml:33]:
```suggestion
          - api: strcmp

```

File: nursery/parse-pe-export-data.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+rule:
+  meta:
+    name: find RVA from a function by parsing the PE Export Data
```
Comment (mr-tz) [inline nursery/parse-pe-export-data.yml:3]:
```suggestion
    name: resolve function by parsing PE exports
```

Comment (mr-tz) [reply nursery/parse-pe-export-data.yml:3]: just an idea to make this more concise

File: nursery/parse-pe-export-data.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    author: sara-rn
+    scope: function
+    examples:
+      - 73CE04892E5F39EC82B00C02FC04C70F
+  features:
+    - and:
+      - os: windows
+      - or:
+        - characteristic: loop
+        - mnemonic: movzx
```
Comment (mr-tz) [inline nursery/parse-pe-export-data.yml:14]: could this also be `mov`?

**Review by mr-tz — CHANGES_REQUESTED** [review]
Nice start, please see the linter output in CI:
```
     (nursery) find RVA from a function by parsing the PE Export Data 
      WARN: filename doesn't match the rule name: Rename rule file to match the rule name, expected: "find-rva-from-a-function-by-parsing-the-pe-export-data.yml", found: "parse-pe-export-data.yml"
      WARN: missing example offset: Add offset of example function
      WARN: EOF format: end file with a single empty line
      WARN: rule format incorrect: use scripts/capafmt.py or adjust as follows
```

File: nursery/parse-pe-export-data.yml (hunk 2 above)
Comment (sara-rn) [reply nursery/parse-pe-export-data.yml:3]: sounds good

File: nursery/parse-pe-export-data.yml (hunk 3 above)
Comment (sara-rn) [reply nursery/parse-pe-export-data.yml:14]: I have seen movzx in 2 samples so far.
mov     eax, [ebp+addressofNamesOrd]
movzx   eax, word ptr [eax+ebx*2]
In the chapter 19 of PMA the same is done with mov: mov  cx, word ptr [ebx+ecx*2]

Comment (sara-rn) [reply nursery/parse-pe-export-data.yml:14]: by removing the movz there will be FP I believe but maybe there is less chance to miss hits, let me know what you think

Comment (mr-tz) [reply nursery/parse-pe-export-data.yml:14]: sorry, I wasn't clear, I meant to suggest to add `mov`?

Comment (williballenthin) [inline nursery/parse-pe-export-data.yml:14]:
```suggestion
        - mnemonic: movzx
        - mnemonic: mov
```

**Comment by mr-tz** [comment]
Thanks!

**Pull request merged**